product_pricelist_per_attribute_value/tests/test_pricelist_per_attribute_value.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2021 Akretion France (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.tests.common import SavepointCase
+
+
+class TestPriceListPerAttributeValue(SavepointCase):
```

Comment:
Test are too complex and their are broken (because the test pass and you do not have the odoo patch applied) so it should failed.

Note : to make it simplier you can only test by calling the method "get_product_price_rule" on product_pricelist (no need of sale order)

Also avoid creating so much pricelist.
In the setup you should
- create one pricelist
- use an existing  product_product_4_product_template Customizable Desk (CONFIG)
- use an existing one like product_category_5 Office Furniture

Then in you test you can

with the Customizable Desk (CONFIG) have for attribute
color : white, black
legs : aluminium, custom, steel
 
test on the two variant
variant_1 = white, aluminum
variant_2 = black, aluminum


test_leg_aluminum(self):
   create one pricelist.item with aluminum
   variant_1 => price with discount
   variant_2 => price with discount

test_leg_aluminum_white(self):
   create one pricelist.item with aluminum and white
   variant_1 => price with discount
   variant_2 => normal price

test_color_white_and_black(self):
   create one pricelist.item with aluminum and white and black
   variant_1 => price with discount
   variant_2 => price with discount

test_leg_aluminum_color_white_and_black(self):
   create one pricelist.item with white and black
   variant_1 => price with discount
   variant_2 => price with discount

Each test should have max 10 lines and it's not necessary to test global, categ or product, choose one case) because you code do not depend on this (it's native odoo code that is already tested that do the job)